File: backend/src/modules/ordering/domain/repositories/order.repository.ts
import type { Order, OrderItem, OrderStatus } from '@modules/ordering/domain';

export const ORDER_REPOSITORY = 'ORDER_REPOSITORY';

export interface OrderRepository {
  createOrder(
    customerName: string,
    items: Omit<OrderItem, 'id'>[],
    totalAmount?: number,
  ): Order;
  updateOrderStatus(
    orderId: string,
    status: OrderStatus,
    message?: string,
  ): boolean;
  getOrder(orderId: string): Order | undefined;
  getAllOrders(): Order[];
  getOrderStats(): any;
}

File: frontend/lib/features/order/domain/repositories/order_repository.dart
import '../../../../core/utils/result.dart';
import '../entities.dart';

/// Repository contract for order operations
abstract class OrderRepository {
  Future<Result<Map<String, dynamic>>> createOrder({
    required String customerName,
    required List<Map<String, dynamic>> items,
  });

  Future<Result<List<Order>>> getAllOrders();

  Future<Result<Order?>> getOrderById(String orderId);

  Future<Result<bool>> updateOrderStatus({
    required String orderId,
    required OrderStatus status,
    String? message,
  });

  Future<Result<List<Order>>> createSampleOrders();

  Future<Result<Map<String, dynamic>?>> getOrderStats();

  Future<Result<bool>> simulateKitchenWorkflow(String? token);
}
